src/binary-encoding.adoc

```diff
@@ -633,43 +628,44 @@ double-precision 3.141592653589793
 If an opcode has a high nibble of `0x6_`, it represents a decimal. Low nibble values `0x_E` and below indicate
 the number of trailing bytes used to encode the decimal.
 
-The body of the decimal is encoded as a <<varint, `VarInt`>> representing its exponent, followed by a `FixedInt`
-representing its coefficient. The width of the coefficient is the total length of the decimal encoding minus the length
-of the exponent. It is possible for the coefficient to have a width of zero, indicating a coefficient of `0`.
+The body of the decimal is encoded as a <<varint, `VarInt`>> representing its coefficient, followed by a `FixedInt`
+representing its exponent. The width of the exponent is the total length of the decimal encoding minus the length
+of the coefficient. It is possible for the exponent to have a width of zero, indicating an exponent of `0`.
 
 Decimal values that require more than 14 bytes can be encoded using the variable-length decimal opcode: `0xF6`.
 
-A decimal with a coefficient of `-0` (which cannot be encoded in a `VarInt`) is encoded using opcode `6F`.
+A decimal with a coefficient of `-0` (which cannot be encoded as a `VarInt`) is encoded using opcode `6F`.
+The opcode is followed by a `VarUInt` representing the byte length and a `FixedInt` representing the exponent.
```

Comment:
I'm guessing this needs to change as well

Reply:
Oh actually I guess it was wrong before (should've read "coefficient ... FixedInt").

But now I suggest this just be a single VarInt, no need for a length prefix.  It'll be more compact and faster!

Reply:
> But now I suggest this just be a single VarInt, no need for a length prefix. It'll be more compact and faster!

Sold!